framework/src/problems/FEProblemBase.C

```diff
@@ -3954,6 +3961,64 @@ FEProblemBase::execute(const ExecFlagType & exec_type)
 
   // Return the current flag to None
   setCurrentExecuteOnFlag(EXEC_NONE);
+
+  if (_uo_aux_state_check && !_checking_uo_aux_state)
+  {
+    // we will only check aux variables and postprocessors
+    // checking more reporter data can be added in the future if needed
+    std::unique_ptr<NumericVector<Number>> x = _aux->currentSolution()->clone();
```

Comment:
Looks like you are allocating and deallocating this vector every time this is called. We could add a vector to aux system to avoid this, but it create another dreaded member in FEProblemBase

Reply:
I agree with @YaqiWang that since this integrity check is default off, if a user turns it on then we should not care too much about performance.